lib/llm/src/local_model.rs

```diff
@@ -126,6 +126,9 @@ impl LocalModel {
         let Some(etcd_client) = endpoint.drt().etcd_client() else {
             anyhow::bail!("Cannot attach to static endpoint");
         };
+        self.ensure_unique(endpoint.component(), self.display_name())
```

Comment:
I _think_ this code path can currently only get hit if dynamo-run is the ingress, and python workers are likely only to get connected to dymamo-run via the new `register_llm` binding which will do all the plumbing to get dynamo-run a ModelDeploymentCard with a display_name here for uniqueness - so I think everything is OK here

But just in case - is there some edge case where you didn't get an MDC and the display_name is an empty string or some default, and then _everything_ would meet this single component name criteria, right? Didn't seem like there was a default value for display_name at a glance, just double checking.

Reply:
I don't think so. It's used in the bindings like you said, in dynamo-run `in=text|batch`, and in the http server. All of those should be fine. I guess we'll find out :-)

There is also some code that will reject empty display names, been in there for a while.